### DISCUSSION TITLE
Magic commands in PowerShell (pwsh)

### DISCUSSION THREAD

**Discussion by wsmelton** [dotnet/interactive#2553] [Q&A]
Should this magic command work with the PowerShell kernel? 

`#!value --name subName --from-value @input:subscription`

![image](https://user-images.githubusercontent.com/11204251/206207399-bd02a96c-8273-4851-87c7-e2cd46216fd7.png)

```
.NET Interactive
© 2020 Microsoft Corporation
Version: 1.0.360204+ffdfe0472eb092844d780bdec9ec7a5bf36ee921
Library version: 1.0.0-beta.22602.4+ffdfe0472eb092844d780bdec9ec7a5bf36ee921
Build date: 2022-12-07T13:59:37.3896977Z
```

**Comment by jonsequitur** [comment]
`#!value` is a separate subkernel and the value you've stored in your example will be stored there, not in the PowerShell kernel. You can read more about that here: https://github.com/dotnet/interactive/blob/main/docs/variable-sharing.md#direct-data-entry-with-value

To directly store a value into a kernel such as the PowerShell kernel, you can use `#!set`:

![image](https://github.com/dotnet/interactive/assets/547415/21df57c9-e2a2-46b5-87af-c33c08381102)

More details here: https://github.com/dotnet/interactive/blob/main/docs/variable-sharing.md#set-a-variable-from-a-value-directly
